enhancements/ingress/set-delete-http-headers.md

```diff
@@ -0,0 +1,1120 @@
+
+Modifying the `host` header on the IngressController therefore could cause HTTP requests to fail with HTTP 503 errors.
+For this reason, the `host` header is in the deny list of headers that are prohibited to be set or deleted using the IngressController API.
+
+Please note that a custom `host` header may be specified via RouteSpec. Please refer to `Open Questions` for more details.
+
+#### The following table summarizes as to which headers are allowed to be customized via RouteSpec or IngresscontrollerSpec
+
+| Header name                | Existing annotation                                 | Configurable on IngressController(Using spec.httpHeaders.actions) | Configurable on Route(Using spec.httpHeaders.actions) | Configurable on IngressController(Using other API or annotation)            | Configurable on Route(Using other API or annotation) |
+|----------------------------|-----------------------------------------------------|-------------------------------------------------------------------|-------------------------------------------------------|-----------------------------------------------------------------------------|------------------------------------------------------|
+| `proxy`                    | None                                                | None                                                              | None                                                  | None                                                                        | None                                                 |
+| `host`                     | None                                                | None                                                              | Yes                                                   | None                                                                        | None                                                 |
+| `strict-transport-security` | `haproxy.router.openshift.io/hsts_header`           | None                                                              | None                                                  | Yes                                                                         | Yes                                                  |                                                     
+| `x-forwarded-for`          | `haproxy.router.openshift.io/set-forwarded-headers` | Yes                                                               | Yes                                                   | Yes(using API spec.httpHeaders.forwardedHeaderPolicy) except delete action) | Yes(using route annotation)                          |
+| `x-ssl`                    | None                                                | Yes                                                               | Yes                                                   | None                                                                        | None                                                 |                    
```

Comment:
I think this table would make more sense under "Risks and Mitigations".

What do you think about combining the "Existing annotation", "Configurable on IngressController(Using other API or annotation)", and "Configurable on Route(Using other API or annotation)" columns?
```suggestion
The following table summarizes which headers are allowed to be customized via the Route API or IngressController API:

| Header name                 | Configurable using IngressController `spec.httpHeaders.actions` | Configurable using Route `spec.httpHeaders.actions` | Configurable using another API                                                                                                               |
|-----------------------------|-----------------------------------------------------------------|-----------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------|
| `proxy`                     | No                                                              | No                                                  | No                                                                                                                                           |
| `host`                      | No                                                              | Yes                                                 | No                                                                                                                                           |
| `strict-transport-security` | No                                                              | No                                                  | `haproxy.router.openshift.io/hsts_header` Route annotation                                                                                   |
| `x-forwarded-for`           | Yes                                                             | Yes                                                 | `haproxy.router.openshift.io/set-forwarded-headers` Route annotation or IngressController `spec.httpHeaders.forwardedHeaderPolicy` API field |
| `x-ssl`                     | Yes                                                             | Yes                                                 | No                                                                                                                                           |
```
| Header name                 | Configurable using IngressController `spec.httpHeaders.actions` | Configurable using Route `spec.httpHeaders.actions` | Configurable using another API                                                                                                               |
|-----------------------------|-----------------------------------------------------------------|-----------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------|
| `proxy`                     | No                                                              | No                                                  | No                                                                                                                                           |
| `host`                      | No                                                              | Yes                                                 | No                                                                                                                                           |
| `strict-transport-security` | No                                                              | No                                                  | `haproxy.router.openshift.io/hsts_header` Route annotation                                                                                   |
| `x-forwarded-for`           | Yes                                                             | Yes                                                 | `haproxy.router.openshift.io/set-forwarded-headers` Route annotation or IngressController `spec.httpHeaders.forwardedHeaderPolicy` API field |
| `x-ssl`                     | Yes                                                             | Yes                                                 | No                                                                                                                                           |

Reply:
yes i did combine this earlier. but @candita wanted me to add more columns from my understanding in slack.

Reply:
This looks better. i am adding this.

Reply:
My point was that for each header that can be set in multiple places, that those places are listed.  [slack](https://redhat-internal.slack.com/archives/C04JJ0T3XT3/p1689281775921699?thread_ts=1689280690.883149&cid=C04JJ0T3XT3)

It would be even better to say which things override others, or order the columns in order of override, if there is any discernible pattern.

What @miciah listed here is still missing the HSTS API in Ingress Operator.

Reply:
@candita we can add it in 

Configurable using another API below haproxy.router.openshift.io/hsts_header Route annotation

Reply:
regarding over-ride I mentioned this in the EP in the `Open Questions` as it is  a common pattern.